projects/packages/forms/src/dashboard/inbox/dataviews/index.js

```diff
@@ -0,0 +1,306 @@
+import { dateI18n } from '@wordpress/date';
+import { useCallback, useMemo, useState } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+/**
+ * Internal dependencies
+ */
+import { store as dashboardStore } from '../../store';
+import InboxResponse from '../response';
+import {
+	viewAction,
+	markAsSpamAction,
+	markAsNotSpamAction,
+	moveToTrashAction,
+	deleteAction,
+	restoreAction,
+} from './actions';
+import { useView, defaultLayouts } from './views';
+
+const EMPTY_ARRAY = [];
+const EMPTY_OBJECT = {};
+const MOBILE_BREAKPOINT = 780;
+const getItemId = item => item.id.toString();
+
+// Function to get the URL of the page or post where the form was submitted.
+const getPath = item => {
+	try {
+		const url = new URL( item.entry_permalink );
+		return url.pathname;
+	} catch {
+		return '';
+	}
+};
+
+/**
+ * Helper function to get the status filter to apply from the URL.
+ * This is the only way to filter the data by `status` as intentionally
+ * we don't want to have a `status` filter in the UI.
+ *
+ * @param {string} urlStatus - The current status from the URL.
+ * @return {string} The status filter to apply.
+ */
+function getStatusFilter( urlStatus ) {
+	// Only allow specific status values.
+	const statusFilter = [ 'inbox', 'spam', 'trash' ].includes( urlStatus ) ? urlStatus : 'inbox';
+	return statusFilter === 'inbox' ? 'draft,publish' : statusFilter;
+}
+
+/**
+ * The DataViews implementation.
+ *
+ * @return {React.ReactElement} The DataViews component.
+ */
+export default function InboxView() {
+	const [ view, setView ] = useView();
+	const [ searchParams, setSearchParams ] = useSearchParams();
+	const [ containerWidth, setContainerWidth ] = useState( 0 );
+	const [ queryArgs, setQueryArgs ] = useState( EMPTY_OBJECT );
+	const containerRef = useResizeObserver(
+		resizeObserverEntries => {
+			setContainerWidth( resizeObserverEntries[ 0 ].borderBoxSize[ 0 ].inlineSize );
+		},
+		{ box: 'border-box' }
+	);
+	const isMobile = containerWidth <= MOBILE_BREAKPOINT;
+	const { setCurrentQuery, setSelectedResponses } = useDispatch( dashboardStore );
+	const selectedResponses = searchParams.get( 'r' );
+	const urlStatus = searchParams.get( 'status' );
+	const statusFilter = getStatusFilter( urlStatus );
+	const filterOptions = useSelect( select => select( dashboardStore ).getFilters(), [] );
+	useEffect( () => {
+		const _filters = view.filters?.reduce( ( accumulator, { field, value } ) => {
+			if ( ! value ) {
+				return accumulator;
+			}
+			if ( field === 'source' ) {
+				accumulator.parent = value;
+			}
+			if ( field === 'date' ) {
+				const [ year, month ] = value.split( '/' ).map( Number );
+				accumulator.after = new Date( Date.UTC( year, month - 1, 1 ) ).toISOString();
+				accumulator.before = new Date( Date.UTC( year, month, 0 ) ).toISOString();
```

Comment:
Before I go down the rabbit hole of WP's date queries, let me just ask the question:

`Date.UTC( y, m, 0 )` returns the date at the last day of month `m` at midnight UTC. So there are still 24 hours left in that month. Does the querying logic take this into account, ignoring the time and only checking for the date (i.e. `y-m-d <=` and not `y-m-d h:m <=`)?

Reply:
Great catch, thank you!

Querying logic uses the whole timestamp so I fixed it [here](https://github.com/Automattic/jetpack/pull/41602/commits/2ab3762af64f12c64e37e34d200bcde6409bbf72).

I tested by updating in console a `feedback` like this:
`wp.data.dispatch('core').saveEntityRecord('postType', 'feedback', {id: 342, date: '2025-02-28T05:00:00.000Z' }, { throwOnError: true })`

Where the `id:342` should be an existing id for each tester (you can check the network request to get it)... After that you should refresh and apply the `February 2025` filter.